plugins/modules/win_zip.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Kento Yagisawa <thel.vadam2485@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_zip
+short_description: Compress file or directory as zip archive on the Windows node
+description:
+- Compress file or directory as zip archive.
+- For non-Windows targets, use the M(ansible.builtin.archive) module instead.
+requirements:
+- PowerShell v5.0 or later
+options:
+  src:
+    description:
+      - File or directory path to be zipped (provide absolute path on the target node).
+    type: path
+    required: yes
+  dest:
+    description:
+      - Destination path of zip file (provide absolute path of zip file on the target node).
+    type: path
+    required: yes
+  overwrite:
```

Comment:
Personally I don't think we should have this. If someone truly wants to always compress the src then they should have a win_file task beforehand to delete it like

```yaml
- win_file:
    path: C:\my\file.zip
    state: absent
```

This allows us to slowly add idempotency to the module that's a lot smarter than just checking if the dest file exists or not.

Reply:
Yeah, I will remove `overwrite` option then.